frontend/app/controllers/MemberOnlyContent.scala

```diff
@@ -1,51 +1,59 @@
-    val accessOpt = ContentAccess.valueOf(membershipAccess)
-    contentApiService.contentItemQuery(Uri.parse(referringContent).path.stripPrefix("/")).map { response =>
-
-      val signInUrl = ((Config.idWebAppUrl / "signin") ? ("returnUrl" -> ("https://theguardian.com/" + referringContent)) ? ("skipConfirmation" -> "true")).toString
+  def membershipContentRedirect = Action { Redirect("/supporter") }
 
-      implicit val countryGroup = UK
+  def membershipContent(referringContentOpt: Option[String] = None) = CachedAction.async { implicit request =>
 
-
-      (for {
-        content <- response.content
+    (for {
+        referringContent <- referringContentOpt
       } yield {
-        if (content.fields.exists(_.membershipAccess.nonEmpty)) {
-          val capiContent: CapiContent = CapiContent(content)
-          val headline: String = capiContent.headline
-          val pageInfo = PageInfo(
-            title = headline,
-            url = request.path,
-            description = capiContent.trailText,
-            customSignInUrl = Some(signInUrl),
-            image = capiContent.mainPicture.map(_.defaultImage)
+          contentApiService.contentItemQuery(Uri.parse(referringContent).path.stripPrefix("/")).map { response =>
+          val signInUrl = ((Config.idWebAppUrl / "signin") ? ("returnUrl" -> ("https://theguardian.com/" + referringContent)) ? ("skipConfirmation" -> "true")).toString
+          implicit val countryGroup = UK
+
+          (for {
+            content <- response.content
+          } yield {
+            if (content.fields.exists(_.membershipAccess.nonEmpty)) {
+              val capiContent: CapiContent = CapiContent(content)
+              val headline: String = capiContent.headline
+              val pageInfo = PageInfo(
+                title = headline,
+                url = request.path,
+                description = capiContent.trailText,
+                customSignInUrl = Some(signInUrl),
+                image = capiContent.mainPicture.map(_.defaultImage)
+              )
+              Ok(views.html.joiner.membershipContent(pageInfo, signInUrl, capiContent, s"Exclusive Members Content: $headline")).
+                withSession(request.session + (DestinationService.JoinReferrer -> ("https://" + Config.guardianHost + "/" + referringContent)))
+            } else {
+              Redirect(("https://theguardian.com/" + referringContent))
+            }
+          }).getOrElse(
+            Redirect(routes.Joiner.tierChooser())
           )
-          Ok(views.html.joiner.membershipContent(pageInfo, accessOpt, signInUrl, capiContent, s"Exclusive Members Content: $headline")).
-            withSession(request.session +  (DestinationService.JoinReferrer -> ("https://" + Config.guardianHost +"/" + referringContent)))
-        } else {
-          Redirect(("https://theguardian.com/" + referringContent))
         }
       }).getOrElse(
```

Comment:
Is it worth considering fold? https://kwangyulseo.com/2014/05/21/scala-option-fold-vs-option-mapgetorelse/